jax/experimental/jax2tf/jax2tf.py

```diff
@@ -854,7 +854,16 @@ def _unexpected_primitive(p: core.Primitive, *args, **kwargs):
 tf_impl[lax.sign_p] = tf.math.sign
 tf_impl[lax.floor_p] = tf.math.floor
 tf_impl[lax.ceil_p] = tf.math.ceil
-tf_impl[lax.round_p] = tf.math.round
+
+def _round(operand):
```

Comment:
I am a bit surprised that we need such a translation, especially the conditional. I am curious what is the bug in the naive translation before? I would have guessed that it is about the sign, but the conditional seems to be a different one.

Reply:
The issue before is, as outline in #4952, JAX rounds away from 0 while TF rounds to the nearest even number.

Reply:
More specifically `jnp.round` rounds towards even (and gets mapped to a soup of HLO and thus to a soup of TF ops instead of `tf.round`) while `lax.round` rounds towards 0.

Reply:
Indeed `jnp.round` should mirror the behavior of `numpy.round`, which is to round towards even, like `tf.round`.
But we are not able to recognize that from the jax2tf side, because `jax2tf` implements conversion only at the primitive level in JAX (i.e. for `lax.round_p`, which is called by `lax.round`; there is no direct conversion for `jnp.round`).

Also to clarify, re. the `lax.round` behaviour, it indeed rounds away from 0 (which you correctly pointed out in the issue :)), see
```python
>>> from jax import lax
>>> import numpy as np
>>> lax.round(np.array([-0.5, -2.5, -1.5, 1.5, 2.5, 0.5], dtype=np.float32))
DeviceArray([-1., -3., -2.,  2.,  3.,  1.], dtype=float32)
```

Reply:
I don't think we'd object if you made the `round` primitive support multiple rounding modes (via a keyword arg) to make it easier to jax2tf to translate to `tf.round` directly in the cases that it applies.

Reply:
Thank you both for explaining. It seems that the round situation is complicated. `jnp.round` is actually implemented much like in this PR, without using `lax.round_p`, so the conversion to TF is irrelevant. Do you think it makes sense to reuse [_round_to_nearest_even](https://github.com/google/jax/blob/1e715121a2ab8020929d9550b7d24e7f36e9ad64/jax/_src/numpy/lax_numpy.py#L1699) for the TF conversion of `lax.round_p`?  

The only use of `lax.round_p` is by way of `lax.round`. I think it makes sense to add a docstring for `lax.round` to explain that this has different behavior than `numpy.round`. 

`lax.round` is used in `jnp.divmod`, which itself has some conditionals preceeding the use.  This means that a conversion of `jnp.divmod` will have both the conditionals from `jnp.divmod` and those from the translation to TF. I do not see an easy way to avoid this, except inventing a mechanism for `jax.custom_jax2tf` to attach to `jnp.divmod`. I do not think it is worth it at this moment.

Reply:
I don't think we can reuse `_round_to_nearest_even` for the TF conversion of `lax.round_p`, since rounding to the nearest even in TF is what we are trying to circumvent (we want to instead round away from 0!).

I think that moving the logic for different rounding strategies inside `round_p` and parameterizing with a kwarg as @hawkinsp suggested is sensible and would circumvent the issue in this case :)

Reply:
I implemented and pushed the suggested change. PTAL whenever you get the chance :)